evalml/automl/automl_search.py

```diff
@@ -577,10 +577,7 @@ def _tune_binary_threshold(self, pipeline, X_threshold_tuning, y_threshold_tunin
             pipeline.threshold = 0.5
             if X_threshold_tuning:
                 y_predict_proba = pipeline.predict_proba(X_threshold_tuning)
-                if isinstance(y_predict_proba, pd.DataFrame):
```

Comment:
Cleanup: we no longer need to worry about whether y_predict_proba is returning np/pd, so no need for this check :D